### PR TITLE
Allow launchpad checklist to display warning icons

### PR DESCRIPTION
**Pull request by dsas**
Some steps might need to show a warning to get attention.

#### Proposed Changes

* Allow launchpad checklist to show a warning for certain steps
* 
#### Testing Instructions

* Checkout this branch
* `yarn run start`
* Go to http://calypso.localhost:3000/setup/newsletter
* Run through the process until you see the launchpad
* Make the `plan_selected` step in `client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/tasks.tsx` have `completed: false, warning: true`
* Refresh the page
* Notice that an exclamation mark is displayed next to the Choose a Plan step

Before | After
-------|------
image | image

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to https://github.com/Automattic/dotcom-forge/issues/1227

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx
Comment (dsas) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx:9]: Maybe warning is the wrong name and this should be `alert` or something.

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/style.scss
Comment (Copons) [inline client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/style.scss]: I wonder if this should be `--color-warning` instead, even though it looks... not great. 🤔 

Some options:

| `-error` | `-warning` | `-warning-light` |
| - | - | - |
| Screenshot 2022-12-02 at 11 32 30 | Screenshot 2022-12-02 at 11 32 18 | Screenshot 2022-12-02 at 11 32 08 |

None of them looks anything like @SaxonF's [suggestion](https://github.com/Automattic/dotcom-forge/issues/1227) (a vivid orange that doesn't seem available in our color vars?), but at least semantically it should be `warning`...

I'd probably go with `warning` and let designers discuss it once they get to test the complete flow.
As far as we are concerned, it's a trivial one-liner fix. 🙂

File: client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx
Comment (Copons) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx:9]: I have literally no opinions on this. 😄 

If I were you, I'd probably keep `warning` just for the sake of "it has the same class as the color!"

Comment (Copons) [reply client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/checklist-item.tsx:9]: Just randomly noticed that GH adds a "Pending" label to my review comment before I submit it, and it has a dark-yellow-ish color like our warnings.

That color is applied to `-warning` and `-attention` classes.
"Attention" seems a good fit, but nobody has ever used it in Calypso, so... yeah, let's go with `warning` and call it a day. No need to block the PR for this.

**Review by Copons — APPROVED** [review]
This works well for me!

I've left a minor comment about changing the icon color, but that really doesn't require a new approval. 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7652835

Thank you @dsas for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7652835).

**Pull request merged**
